client/mm/mm_simple_arb.go

```diff
@@ -73,20 +79,97 @@ type simpleArbMarketMaker struct {
 	host    string
 	baseID  uint32
 	quoteID uint32
-	cex     libxc.CEX
+	cex     cex
 	// cexTradeUpdatesID is passed to the Trade function of the cex
 	// so that the cex knows to send update notifications for the
 	// trade back to this bot.
-	cexTradeUpdatesID int
-	core              clientCore
-	log               dex.Logger
-	cfg               *SimpleArbConfig
-	mkt               *core.Market
-	book              dexOrderBook
-	rebalanceRunning  atomic.Bool
+	core             clientCore
+	log              dex.Logger
+	cfg              *SimpleArbConfig
+	mkt              *core.Market
+	book             dexOrderBook
+	rebalanceRunning atomic.Bool
 
 	activeArbsMtx sync.RWMutex
 	activeArbs    []*arbSequence
+
+	// If pendingBaseRebalance/pendingQuoteRebalance are true, it means
+	// there is a pending deposit/withdrawal of the base/quote asset,
+	// and no other deposits/withdrawals of that asset should happen
+	// until it is complete.
+	pendingBaseRebalance  atomic.Bool
+	pendingQuoteRebalance atomic.Bool
+}
+
+// rebalanceAsset checks if the balance of an asset on the dex and cex are
+// below the minimum amount, and if so, deposits or withdraws funds from the
+// CEX to make the balances equal. If it is not possible to bring both the DEX
+// and CEX balances above the minimum amount, no action will be taken.
+func (a *simpleArbMarketMaker) rebalanceAsset(base bool) {
+	var assetID uint32
+	var minAmount uint64
+	if base {
+		assetID = a.baseID
+		minAmount = a.cfg.MinBaseAmt
+	} else {
+		assetID = a.quoteID
+		minAmount = a.cfg.MinQuoteAmt
+	}
+
+	dexBalance, err := a.core.AssetBalance(assetID)
+	if err != nil {
+		a.log.Errorf("Error getting asset %d balance: %v", assetID, err)
+		return
+	}
+
+	cexBalance, err := a.cex.Balance(assetID)
+	if err != nil {
+		a.log.Errorf("Error getting asset %d balance on cex: %v", assetID, err)
+		return
+	}
+
+	if (dexBalance.Available+cexBalance.Available)/2 < minAmount {
+		a.log.Warnf("Cannot rebalance asset %d because balance is too low on both DEX and CEX", assetID)
+		return
+	}
+
+	var requireDeposit bool
+	if cexBalance.Available < minAmount {
+		requireDeposit = true
+	} else if dexBalance.Available >= minAmount {
+		// No need for withdrawal or deposit.
+		return
+	}
+
+	onConfirm := func() {
+		if base {
+			a.pendingBaseRebalance.Store(false)
+		} else {
+			a.pendingQuoteRebalance.Store(false)
+		}
+	}
+
+	if requireDeposit {
+		amt := (dexBalance.Available+cexBalance.Available)/2 - cexBalance.Available
+		err = a.cex.Deposit(a.ctx, assetID, amt, onConfirm)
```

Comment:
``` golang
if (dexBalance.Available+cexBalance.Available)/2 < minAmount {
a.log.Warnf("Cannot rebalance asset %d because balance is too low on both DEX and CEX", assetID)
return
}
```

This check above ensures that dex balance > cex balance, (or vice-versa for the withdrawal). You're right about the edge behavior though.. if the minimum balance is 100, the DEX has 99, and the CEX has 102, we probably shouldn't be withdrawing 1.5.

Reply:
Added `MinBaseTransfer` and `MinQuoteTransfer`.